### ISSUE TITLE
Is pytest-steps compatible with async function?

### ISSUE THREAD

**Issue by minzoovv** [smarie/python-pytest-steps#33]
I found the pytest-steps and I'm trying to use it my code,
but some functions in my code is async function, so I'm curious about that this module is compatible with pytest.mark.asyncio or other async functions! 

I'll wait your comment and have a nice day :)

**Comment by smarie** [comment]
Thanks for reaching out ! Any feedback is always great

I have not tested this plugin with async functions. However I do not see any reason why this would not be compliant, as long as the **test functions** are normal functions (or generators if you use the new syntax with `yield` statements, ). 

Let me know if it does not work for you - in that case please send a "minimum reproducible example"  of the errors you are facing.

**Comment by smarie** [comment]
Closing this issue as it seems solved. Please reopen if this is not the case.

**Comment by Vernadskii** [comment]
[There is no way](https://stackoverflow.com/questions/55647753/call-async-function-from-sync-function-while-the-synchronous-function-continues) to run async functions in sync function. As I see your decorators type on default is sync. I guess it's not compatible at all :c

**Comment by smarie** [comment]
good point @VernadskyDanya : I guess that `pytest.mark.asyncio` makes pytest run this async directly in the event loop, so the pytest-steps decorator which creates a sync function wrapper, is not compliant. Thanks for the precision!
